sdk/python/tests/unit/online_store/test_online_retrieval.py

```diff
@@ -274,6 +275,299 @@ def test_get_online_features() -> None:
+                        "age": ValueProto(int64_val=3),
+                    },
+                    _utc_now(),
+                    _utc_now(),
+                )
+            ],
+            progress=None,
+        )
+
+        customer_key = EntityKeyProto(
+            join_keys=["customer_id", "driver_id"],
+            entity_values=[ValueProto(string_val="5"), ValueProto(int64_val=1)],
+        )
+        provider.online_write_batch(
+            config=store.config,
+            table=customer_driver_combined_fv,
+            data=[
+                (
+                    customer_key,
+                    {"trips": ValueProto(int64_val=7)},
+                    _utc_now(),
+                    _utc_now(),
+                )
+            ],
+            progress=None,
+        )
+
+        assert len(store.list_entities()) == 3
+        assert len(store.list_entities(tags=TAGS)) == 2
+
+        # Retrieve two features using two keys, one valid one non-existing
+        result = store.get_online_features(
+            features=[
+                "driver_locations:lon",
+                "customer_profile:avg_orders_day",
+                "customer_profile:name",
+                "customer_driver_combined:trips",
+            ],
+            entity_rows=[
+                {"driver_id": 1, "customer_id": "5"},
+                {"driver_id": 1, "customer_id": 5},
+            ],
+            full_feature_names=False,
+        ).to_dict()
+
+        assert "lon" in result
+        assert "avg_orders_day" in result
+        assert "name" in result
+        assert result["driver_id"] == [1, 1]
+        assert result["customer_id"] == ["5", "5"]
+        assert result["lon"] == ["1.0", "1.0"]
+        assert result["avg_orders_day"] == [1.0, 1.0]
+        assert result["name"] == ["John", "John"]
+        assert result["trips"] == [7, 7]
+
+        # Ensure features are still in result when keys not found
+        result = store.get_online_features(
+            features=["customer_driver_combined:trips"],
+            entity_rows=[{"driver_id": 0, "customer_id": 0}],
+            full_feature_names=False,
+        ).to_dict()
+
+        assert "trips" in result
+
+        result = store.get_online_features(
+            features=["customer_profile_pandas_odfv:on_demand_age"],
+            entity_rows=[{"driver_id": 1, "customer_id": "5"}],
+            full_feature_names=False,
+        ).to_dict()
+
+        assert "on_demand_age" in result
+        assert result["driver_id"] == [1]
+        assert result["customer_id"] == ["5"]
+        assert result["on_demand_age"] == [4]
+
+        # invalid table reference
+        with pytest.raises(FeatureViewNotFoundException):
+            store.get_online_features(
+                features=["driver_locations_bad:lon"],
+                entity_rows=[{"driver_id": 1}],
+                full_feature_names=False,
+            )
+
+        # TODO: Need to fix these tests to actually run corecctly.
```

Comment:
does this needs to revisit now or later?

Reply:
Was going to handle that in a follow up PR too. Trying to get a release out soon was the main reason.

Reply:
Removed.